src/modules/workspaces/guards/workspaces.guard.ts

```diff
@@ -57,13 +64,22 @@ export class WorkspaceGuard implements CanActivate {
       );
     }
 
+    let verified = false;
+    const workspaceLogAction =
+      this.reflector.get('workspaceLogAction', context.getHandler()) || null;
+
     if (accessContext === AccessContext.WORKSPACE) {
-      return this.verifyWorkspaceAccessByRole(user, id, requiredRole);
+      verified = await this.verifyWorkspaceAccessByRole(user, id, requiredRole);
     } else if (accessContext === AccessContext.TEAM) {
-      return this.verifyTeamAccessByRole(user, id, requiredRole);
+      verified = await this.verifyTeamAccessByRole(user, id, requiredRole);
+    }
+
+    if (verified && workspaceLogAction === WorkspaceLogType.DELETE_ALL) {
```

Comment:
I do not understand DELETE_ALL usage, neither the getItems request. There is no DELETE_ALL log whatsoever contemplated on the product request